### PR TITLE
Added CompoundIdentifier class  is_input_header negative test

### PR DESCRIPTION
**Pull request by musasizivictoria**
Thank you for taking your time to contribute to Ersilia, just a few checks before we proceed
- [x] Have you followed the guidelines in our [Contribution Guide](https://github.com/ersilia-os/ersilia/blob/master/CONTRIBUTING.md)
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

**Description**
Added negative tests for the is_input_header method in the CompoundIdentifier class. These tests ensure that invalid headers are correctly identified by the method and return False.

**Changes to be made**

1. Implement negative tests to check the method's behavior for invalid headers 

**Status**

- [x] Implemented and successfully ran negative tests locally with all tests passing

**To do**

- [x] Review and ensure code coverage
- [x] Continue with further unit tests for other methods of the CompoundIdentifier class

Is this pull request related to any open issue? If yes, replace issueID below with the issue ID

Related to [#<issueID> ](https://github.com/ersilia-os/ersilia/issues/1319)

### CODE REVIEW

**Comment by DhanshreeA** [comment]
@musasizivictoria could you please resolve conflicts? Thanks!

**Comment by musasizivictoria** [comment]
sorry @DhanshreeA  it seems we are working on same thng many people

File: test/test_compound_identifier.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ def test_is_input_header_positive(compound_identifier, header):
     """Test that valid input headers return True."""
     assert compound_identifier.is_input_header(header) is True
 
-@pytest.mark.parametrize("header", ["key", "inchiKey", "KEY", "INCHIKEY"])
```
Comment (DhanshreeA) [inline test/test_compound_identifier.py:13]: Hey @musasizivictoria please don't remove this test. You should add this test after your `input_header_negative` test in this PR.

**Review by DhanshreeA — CHANGES_REQUESTED** [review]

File: test/test_compound_identifier.py (hunk 1 above)
Comment (musasizivictoria) [reply test/test_compound_identifier.py:13]: so sorry @DhanshreeA  , it was an oversight on method names

Comment (musasizivictoria) [reply test/test_compound_identifier.py:13]: Have resolved it i guess

Comment (musasizivictoria) [reply test/test_compound_identifier.py:13]: am ready to continue with other tests  once this is approved

**Review by DhanshreeA — APPROVED** [review]

**Comment by DhanshreeA** [comment]
@musasizivictoria could you please take care of the conflicts?

**Comment by musasizivictoria** [comment]
> @musasizivictoria could you please take care of the conflicts?

Addressed, thanks @DhanshreeA

**Comment by musasizivictoria** [comment]
Also some clarity here if you don't mind
https://github.com/ersilia-os/ersilia/issues/1319#issuecomment-2426081972

**Pull request merged**